extensions/cornerstone/src/services/ViewportActionCornersService/ViewportActionCornersService.ts

```diff
@@ -51,28 +49,115 @@ class ViewportActionCornersService extends PubSubService {
     if (addComponentsImplementation) {
       this.serviceImplementation._addComponents = addComponentsImplementation;
     }
-    if (clearComponentsImplementation) {
-      this.serviceImplementation._clear = clearComponentsImplementation;
+    if (clearImplementation) {
+      this.serviceImplementation._clear = clearImplementation;
+    }
+    if (lockItemImplementation) {
+      this.serviceImplementation._lockItem = lockItemImplementation;
+    }
+    if (unlockItemImplementation) {
+      this.serviceImplementation._unlockItem = unlockItemImplementation;
+    }
+    if (toggleLockImplementation) {
+      this.serviceImplementation._toggleLock = toggleLockImplementation;
+    }
+    if (isItemLockedImplementation) {
+      this.serviceImplementation._isItemLocked = isItemLockedImplementation;
+    }
+    if (showItemImplementation) {
+      this.serviceImplementation._showItem = showItemImplementation;
+    }
+    if (hideItemImplementation) {
+      this.serviceImplementation._hideItem = hideItemImplementation;
+    }
+    if (toggleVisibilityImplementation) {
+      this.serviceImplementation._toggleVisibility = toggleVisibilityImplementation;
+    }
+    if (isItemVisibleImplementation) {
+      this.serviceImplementation._isItemVisible = isItemVisibleImplementation;
+    }
+    if (openItemImplementation) {
+      this.serviceImplementation._openItem = openItemImplementation;
+    }
+    if (closeItemImplementation) {
+      this.serviceImplementation._closeItem = closeItemImplementation;
+    }
+    if (closeAllItemsImplementation) {
+      this.serviceImplementation._closeAllItems = closeAllItemsImplementation;
+    }
+    if (isItemOpenImplementation) {
+      this.serviceImplementation._isItemOpen = isItemOpenImplementation;
     }
   }
 
   public getState() {
     return this.serviceImplementation._getState();
   }
 
-  public addComponent(component: ActionComponentInfo) {
+  public addComponent(component: Types.ActionComponentInfo) {
     this.serviceImplementation._addComponent(component);
   }
 
-  public addComponents(components: Array<ActionComponentInfo>) {
+  public addComponents(components: Array<Types.ActionComponentInfo>) {
     this.serviceImplementation._addComponents(components);
   }
 
-  public clear(viewportId: string) {
+  public clear(viewportId: string): void {
     this.serviceImplementation._clear(viewportId);
   }
 
-  public getAlignAndSide(location: ViewportActionCornersLocations): AlignAndSide {
+  /* lock / unlock */
+  public lockItem(viewportId: string, itemId: string): void {
```

Comment:
Since you are adding a whole bunch of items, can you make the new values just public unlockItem: (viewportId,itemId) => void;
and assign unlockItem directly from the set method?

Reply:
I don't understand lets discuss tomorrow

Reply:
I will do this for a PR for all services